src/cpp/common/tls_credentials_options.cc

```diff
@@ -274,6 +294,11 @@ TlsCredentialsOptions::TlsCredentialsOptions(
   }
 }
 
+/** Whenever a TlsCredentialsOptions instance is created, the caller takes
+ *  ownership of the c_credentials_options_ pointer (see e.g. the implementation
+ *  of the TlsCredentials API in secure_credentials.cc). For this reason, the
+ *  TlsCredentialsOptions destructor is not responsible for freeing
+ *  c_credentials_options_. **/
```

Comment:
Since `c_credentials_options_` is created inside `TlsCredentialsOptions`constructor I think it also makes sense to destruct it in the destructor of `TlsCredentialsOptions`. Most of all, a caller does not know about `c_credentials_options_` right?

Reply:
That's true, but every current caller of `TlsCredentialsOptions` immediately takes ownership of `c_credentials_options_`. For that reason, if we try to delete `c_credentials_options_` in the destructor, then we will get an error.

Reply:
I see. Can you please specify what are the callers of `TlsCredentialsOptions`? Do we have any callers other than users that create spiffe credentials?

Reply:
Outside of tests, the only callers are the `TlsCredentials` and `TlsServerCredentials` API's that are found in the files credentials.h, credentials_impl.h, server_credentials.h, and server_credentials_impl.h.

And yes, I can add this discussion and a list of the callers in the tls_credentials_options.h file where the destructor is exposed.

Reply:
I see. Let's discuss more about it next Monday once you are at the office :)